core/server/models/base/index.js

```diff
@@ -19,10 +19,12 @@ var _          = require('lodash'),
     validation = require('../../data/validation'),
     plugins    = require('../plugins'),
     i18n       = require('../../i18n'),
-
     ghostBookshelf,
     proto;
 
+// event connector
+require('./when');
```

Comment:
name change: 
there was actually no a name change, i decided to go with the `when` naming, which i invented in #6861 . i think this is preference. I decided against the naming `events`, because calling a file/module `events` in JS sounds like the file offers a way to send events, which is not the case. An alternative naming could be `listeners` or ....?

moving code: 
i initiated the `when` module already in #6861  inside `/models/base/index.js`, because the file (`when`) is part of the `base` folder. The reason why it is maybe a bit confusing, is that our `index.js` are often misused in our code base, because an `index.js` brings the power of exporting a bigger module and can contain some smaller bootstrap code). So technically it belongs into this file.

Let me initialise the `when` module in `models/index.js`, but i would love to see getting rid of implementing classes, models, logic inside `index.js` files. (this has also a big disadvantage when writing tests, because naming a unit test file always `index_spec.js` brings confusion and for example problems with searching for files in the IDE.